openssl/src/bn.rs

```diff
@@ -868,13 +868,46 @@ impl BigNumRef {
     /// assert_eq!(BigNum::from_slice(&s_vec).unwrap(), r);
     /// ```
     pub fn to_vec(&self) -> Vec<u8> {
-        let size = self.num_bytes() as usize;
-        let mut v = Vec::with_capacity(size);
+        let mut v = vec![0u8; self.num_bytes() as usize];
+        self.to_slice(&mut v).unwrap();
+        v
+    }
+
+    /// Encodes the absolute value of `self` into a big-endian buffer.
+    ///
+    /// `self` can be recreated by using `from_slice`.
+    ///
+    /// This method fails if the buffer is not large enough. If the
+    /// buffer is larger than necessary, the number is prefixed by
+    /// zeroes so that the result is always a valid big-endian number.
+    ///
+    /// ```
+    /// # use openssl::bn::BigNum;
+    /// let s = -BigNum::from_u32(4543).unwrap();
+    /// let r = BigNum::from_u32(4543).unwrap();
+    ///
+    /// let mut b = [0u8; 4];
+    /// s.to_slice(&mut b).unwrap();
+    /// assert_eq!(BigNum::from_slice(&b).unwrap(), r);
+    ///
+    /// let mut b = [0u8; 1];
+    /// assert_eq!(s.to_slice(&mut b), Err(()));
+    /// ```
+    pub fn to_slice(&self, buf: &mut [u8]) -> Result<(), ()> {
```

Comment:
This can just assert that the buffer is big enough.

Reply:
We could do this, yes. But then we'd write the same code again in `TryFrom` (see #1203). It also doesn't seem like the right thing to do for security to force the consumer of this API to check for theoretical errors before calling it to avoid an abort. This is particularly true when the incoming `BigNum` data is often user or network input.

Reply:
@sfackler Do you have any interest in merging support for `TryFrom` integer conversions? Your crate is the only one who can supply this interface because of Rust's impl rules.

Reply:
Sure, that seems reasonable.